v2/internal/controllers/crd_dataprotection_backupinstance_test.go

```diff
@@ -0,0 +1,302 @@
+				ARMID: fmt.Sprintf("/subscriptions/%s/providers/Microsoft.Authorization/roleDefinitions/acdd72a7-3385-48ef-bd42-f606fba81ae7", tc.AzureSubscription), // This is Reader Role
+			},
+		},
+	}
+
+	//create backup instance
+	biName := "asotestbackupinstance"
+	backupInstance := &dataprotection.BackupVaultsBackupInstance{
+		ObjectMeta: tc.MakeObjectMeta(biName),
+		Spec: dataprotection.BackupVaults_BackupInstance_Spec{
+			Owner: testcommon.AsOwner(backupVault),
+			Properties: &dataprotection.BackupInstance{
+				ObjectType:   to.Ptr("BackupInstance"),
+				FriendlyName: to.Ptr(biName),
+				DataSourceInfo: &dataprotection.Datasource{
+					ObjectType:        			to.Ptr("Datasource"),
+					DatasourceType:    			to.Ptr(cluster.GetType()),
+					ResourceUriReference:       tc.MakeReferenceFromResource(cluster),
+					ResourceName:      			to.Ptr(cluster.AzureName()),
+					ResourceLocation:  			cluster.Spec.Location,
+					ResourceType:      			to.Ptr(cluster.GetType()),
+					ResourceReference: 			tc.MakeReferenceFromResource(cluster),
+				},
+				DataSourceSetInfo: &dataprotection.DatasourceSet{
+					ObjectType:        			to.Ptr("DatasourceSet"),
+					DatasourceType:    			to.Ptr(cluster.GetType()),
+					ResourceUriReference:       tc.MakeReferenceFromResource(cluster),
+					ResourceName:      			to.Ptr(cluster.AzureName()),
+					ResourceType:      			to.Ptr(cluster.GetType()),
+					ResourceLocation:  			cluster.Spec.Location,
+					ResourceReference: 			tc.MakeReferenceFromResource(cluster),
+				},
+				PolicyInfo: &dataprotection.PolicyInfo{
+					PolicyReference: tc.MakeReferenceFromResource(backupPolicy),
+					PolicyParameters: &dataprotection.PolicyParameters{
+						DataStoreParametersList: []dataprotection.DataStoreParameters{
+							{
+								AzureOperationalStoreParameters: &dataprotection.AzureOperationalStoreParameters{
+									DataStoreType:   to.Ptr(dataprotection.AzureOperationalStoreParameters_DataStoreType_OperationalStore),
+									ResourceGroupReference: tc.MakeReferenceFromResource(rg),
+									ObjectType:      to.Ptr(dataprotection.AzureOperationalStoreParameters_ObjectType_AzureOperationalStoreParameters),
+								},
+							},
+						},
+						BackupDatasourceParametersList: []dataprotection.BackupDatasourceParameters{
+							{
+								KubernetesCluster: &dataprotection.KubernetesClusterBackupDatasourceParameters{
+									SnapshotVolumes:              to.Ptr(true),
+									IncludeClusterScopeResources: to.Ptr(true),
+									ObjectType:                   to.Ptr(dataprotection.KubernetesClusterBackupDatasourceParameters_ObjectType_KubernetesClusterBackupDatasourceParameters),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tc.CreateResourcesAndWait(cluster, acct, blobService, blobContainer, backupVault, backupPolicy, extension,
+		trustedAccessRoleBinding, extenstionRoleAssignment, clusterRoleAssignment, clusterMSIRoleAssignment, snapshotRGRoleAssignment, backupInstance)
```

Comment:
Our coding standard is to have all parameters on one line, or one per line, to make it easier to scan through the list.

```suggestion
	tc.CreateResourcesAndWait(
		cluster, 
		acct, 
		blobService, 
		blobContainer, 
		backupVault, 
		backupPolicy, 
		extension,
		trustedAccessRoleBinding, 
		extenstionRoleAssignment, 
		clusterRoleAssignment, 
		clusterMSIRoleAssignment, 
		snapshotRGRoleAssignment, 
		backupInstance)
```

Reply:
It's good to have all of these in one `CreateResources()` call , ensuring that ASO correctly manages the dependencies.

Reply:
changes indentation